src/Chapter3.hs

```diff
@@ -1006,6 +1149,50 @@ Implement data types and typeclasses, describing such a battle between two
 contestants, and write a function that decides the outcome of a fight!
 -}
 
+data BattleResult = Victory | Defeat
+
+class Fighter x y where
+    attackEm :: x -> y -> y
+    runAway :: x -> y -> BattleResult
```

Comment:
It would be enough to make the instance for each creature separately:
For that you can use constraints inside the typeclass as well:

```suggestion
class Fighter x where
    attackEm :: (Fighter y) => x -> y -> y
    runAway :: (Fighter y) => x -> y -> BattleResult
```